### ISSUE TITLE
Compile Error

### ISSUE THREAD

**Issue by Bizes** [Tewesday/RealmDemo#2]
Hi,

I don't know how active the development of this project is, but I am interested in looking at how things work in the project.

I tried to clone and start the project on my computer but I can't compile the project.
The errors I am getting are almost all like this:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2075	'_Elems': initialization requires a brace-enclosed initializer list	RealmDemo	C:\Users\Admin\Documents\Unreal Projects\RealmDemo\Source\RealmDemo\Generic\OctreeSys.h	54	
```

So it seems to be something with the Octree BoundingBox.
For example:
auto SearchBox = OrthoTree::BoundingBox2D({ Position.X - Radius, Position.Y - Radius }, { Position.X + Radius, Position.Y + Radius });

My guess is that it is something with the array initialization. But I couldn't find a solution and my knowledge here is quite limited. Maybe someone has an idea of how to fix it?
I tried to initialize the two arrays independently and pass them as arguments to the BoundingBox constructor but that did not work.

All the best

**Comment by Tewesday** [comment]
Hello,

Thanks for letting me know about this. I want to make sure this project still builds successfully even though I'm mostly leaving it alone for now.

I just finished replacing the third party octree with an unreal octree. It's something I wanted to do anyway but I had the third party version working so I hadn't done it yet.

It should now compile with the octree change but I need to figure out why the MassEntities/Actors are not spawning fully now.

**Comment by Tewesday** [comment]
The spawning issue I mentioned seems to be a Linux/UE 5.2 issue.

Everything seems to be working with UE 5.1 on windows.

**Comment by Bizes** [comment]
Hi thank you for the quick response. 
I pulled the changes and everything seems to work!

Thanks again!

May I ask if there was a particular reason that you left the project as is? Have you done all the testing you wanted or did you move to another approach? Or was it something completely unrelated.

**Comment by Tewesday** [comment]
Glad to hear it's working for you now.

My goal with this project was to do a rough implementation of basic RTS with Mass. I feel like I have accomplished that (although some code cleanup could be in order). 

I think Mass is great and having first party support for an ECS is very important, but it's still got some work ahead to be production ready in my opinion. 

In the interest of development time for a project like this I decided to switch to FLECS as it's a bit more mature and feature complete.
